src/lib/wallet/IndyWallet.ts

```diff
@@ -32,6 +33,22 @@ export class IndyWallet implements Wallet {
     }
 
     this.wh = await this.indy.openWallet(this.walletConfig, this.walletCredentials);
+
+    try {
+      logger.log(`Creating master secret...`);
+      this.masterSecretId = await this.indy.proverCreateMasterSecret(this.wh, this.walletConfig.id);
+    } catch (error) {
+      logger.log('error', error);
+      if (error.indyName && error.indyName === 'AnoncredsMasterSecretDuplicateNameError') {
+        // master secret id is the same as the master secret id passed in the create function
+        // so if it already exists we can just assign it.
+        this.masterSecretId = this.walletConfig.id;
+        logger.log(`master secret with id ${this.masterSecretId} already exists`, error.indyName);
+      } else {
+        throw error;
+      }
+    }
```

Comment:
I'd like to make a generic approach for calling indy functions that are allowed to fail. As this snippet is exactly the. same as the snippet above that tries to create a wallet that possibly already exists. But that's something for a future PR I think

Reply:
That's right and we don't have to hurry with this. I decided to leave it and wait for more cases like that to create better abstraction afterward.